commands/acr-build.ts

```diff
@@ -0,0 +1,116 @@
+// The user is then asked to name & tag the image. A build is queued for the image in the selected registry.
+// Selected source code must contain a path to the desired dockerfile.
+export async function queueBuild(dockerFileUri?: vscode.Uri): Promise<void> {
+    status.appendLine("Obtaining Subscription and Client");
+    let subscription = await acquireSubscription();
+    let client = AzureUtilityManager.getInstance().getContainerRegistryManagementClient(subscription);
+    const resourceGroupClient = new ResourceManagementClient(AzureUtilityManager.getInstance().getCredentialByTenantId(subscription.tenantId), subscription.subscriptionId);
+
+    let resourceGroup = await acquireResourceGroup(subscription, resourceGroupClient);
+    let resourceGroupName = resourceGroup.name;
+
+    let registry: Registry = await quickPickACRRegistry(subscription, resourceGroupName);
+    let registryName = registry.name;
+
+    let folder: vscode.WorkspaceFolder;
+    if (vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders.length === 1) {
+        folder = vscode.workspace.workspaceFolders[0];
+    } else {
+        folder = await (<any>vscode).window.showWorkspaceFolderPick();
+    }
+    let sourceLocation: string = folder.uri.path;
+
+    let relativeDockerPath = 'Dockerfile';
+    if (dockerFileUri.path.indexOf(sourceLocation) !== 0) {
+        //Currently, there is no support for selecting source location folders that don't contain a path to the triggered dockerfile.
+        throw new TypeError("Source code path must be a parent of the Dockerfile path")
+    } else {
+        relativeDockerPath = dockerFileUri.path.toString().substring(sourceLocation.length);
+    }
+
+    // Prompting for name so the image can then be pushed to a repository.
+    const opt: vscode.InputBoxOptions = {
+        prompt: 'Image name and tag in format  <name>:<tag>',
+    };
+    const name: string = await vscode.window.showInputBox(opt);
+
+    /* tslint:disable-next-line:insecure-random */
+    let id = Math.floor(Math.random() * Math.pow(10, idPrecision));
+    status.appendLine("Setting up temp file with 'sourceArchive" + id + ".tar.gz' ");
+    let tarFilePath = url.resolve(os.tmpdir(), `sourceArchive${id}.tar.gz`);
+
+    status.appendLine("Uploading Source Code to " + tarFilePath);
+    sourceLocation = await uploadSourceCode(client, registryName, resourceGroupName, sourceLocation, tarFilePath);
+
+    status.appendLine("Setting up Build Request");
+    let buildRequest: QuickBuildRequest = {
+        'type': 'QuickBuild',
+        'imageNames': [name],
+        'isPushEnabled': true,
+        'sourceLocation': sourceLocation,
+        'platform': { 'osType': 'Linux' },
+        'dockerFilePath': relativeDockerPath
+    };
+    status.appendLine("Queueing Build");
+    try {
+        await client.registries.queueBuild(resourceGroupName, registryName, buildRequest);
+    } catch (error) {
```

Comment:
Check with @estebanreyl  for the better solution to handle error message.

Reply:
removed try-catches